app/addons/documents/queryoptions/stores.js

```diff
@@ -211,7 +211,7 @@ function (app, FauxtonAPI, ActionTypes) {
           params.end_key = betweenKeys.endkey;
         }
       } else if (this._showByKeys) {
-        params.keys = this._byKeys;
+        params.keys = this._byKeys.replace(/\n/g, '');
```

Comment:
that probably won't work for windows `\r\n` newlines

untested alternative: `/\r?\n/`